src/cascadia/TerminalControl/SearchBoxControl.cpp

```diff
@@ -209,4 +224,143 @@ namespace winrt::Microsoft::Terminal::Control::implementation
+        }
+
+        if (totalMatches > MaximumTotalResultsToShowInStatus)
+        {
+            totalString = TotalResultsTooHighStatus;
+        }
+        else
+        {
+            totalString = fmt::format(L"{}", totalMatches);
+        }
+
+        return winrt::hstring{ fmt::format(RS_(L"TermControl_NumResults").c_str(), currentString, totalString) };
+    }
+
+    // Method Description:
+    // - Helper method to measure the width of the text block given the text and the font size
+    // Arguments:
+    // - text: the text to measure
+    // - fontSize: the size of the font to measure
+    // Return Value:
+    // - the size in pixels
+    double SearchBoxControl::_TextWidth(winrt::hstring text, double fontSize)
+    {
+        winrt::Windows::UI::Xaml::Controls::TextBlock t;
+        t.FontSize(fontSize);
+        t.Text(text);
+        t.Measure({ FLT_MAX, FLT_MAX });
+        return t.ActualWidth();
+    }
+
+    // Method Description:
+    // - This method tries to predict the maximal size of the status box
+    // by measuring different possible statuses
+    // Return Value:
+    // - the size in pixels
+    double SearchBoxControl::_GetStatusMaxWidth()
+    {
+        const auto fontSize = StatusBox().FontSize();
+        const auto maxLength = std::max({ _TextWidth(_FormatStatus(-1, -1), fontSize),
+                                          _TextWidth(_FormatStatus(0, -1), fontSize),
+                                          _TextWidth(_FormatStatus(MaximumTotalResultsToShowInStatus, MaximumTotalResultsToShowInStatus - 1), fontSize),
+                                          _TextWidth(_FormatStatus(MaximumTotalResultsToShowInStatus + 1, MaximumTotalResultsToShowInStatus - 1), fontSize),
+                                          _TextWidth(_FormatStatus(MaximumTotalResultsToShowInStatus + 1, MaximumTotalResultsToShowInStatus), fontSize) });
+
+        return maxLength;
+    }
+
+    // Method Description:
+    // - Formats and sets the status message in the status box.
+    // Increases the size of the box if required.
+    // Arguments:
+    // - totalMatches - total number of matches (search results)
+    // - currentMatch - the index of the current match (0-based)
+    // Return Value:
+    // - <none>
+    void SearchBoxControl::SetStatus(int32_t totalMatches, int32_t currentMatch)
+    {
+        const auto status = _FormatStatus(totalMatches, currentMatch);
+        const auto requiredWidth = _TextWidth(status, StatusBox().FontSize());
+        if (requiredWidth > StatusBox().Width())
+        {
+            StatusBox().Width(requiredWidth);
+        }
+
+        StatusBox().Text(status);
+    }
+
+    // Method Description:
+    // - Enables / disables results navigation buttons
+    // Arguments:
+    // - enable: if true, the buttons should be enabled
+    // Return Value:
+    // - <none>
+    void SearchBoxControl::SetNavigationEnabled(bool enabled)

```

Comment:
nit: this is a property setter! also, `enter` is a way to navigate, but it still works while this is false i bet...

Reply:
yup! confirmed in TextBoxKeyDown

Reply:
this is still a "property setter"; should it be modelled as such in the IDL?